signal/signal.go

```diff
@@ -111,22 +111,28 @@ type Interceptor struct {
 	// close this channel.
 	quit chan struct{}
 
+	// exitCode represents a channel for receiving exit codes.
+	// It is used to communicate the exit status when the main
+	// interrupt handler exits.
+	exitCode chan int
```

Comment:
why use a channel for this? channels for the rest make sense to me cause other sub-systems might wait on these but we would not wait on a specific exit code to be set. 

I think this can just be a mutex protected int and probably need a bool to indicate that we have in fact exited... ie, GetExitCode can return an int along with a bool. If the bool is false, it means we have not exited yet.